### ISSUE TITLE
Display friendly error message when spawning CGI process fails

### ISSUE THREAD

**Issue by GoogleCodeExporter** [cztomczak/phpdesktop#127]
See https://github.com/cztomczak/phpdesktop/wiki/Knowledge-Base#error-500-internal-server-error-cannot-spawn-cgi-process-cdocuments-and-settingswwwindexphp-no-such-device-or-address

If you try to run phpdesktop with PHP 5.6 on Windows XP you will get error message like this:

```
  Error 500: Internal Server Error
  Cannot spawn CGI process [C:\...\phpdesktop\www/index.php]: No such device or address
```

This can be fixed similarly as in Issue #131 by using CefLoadHandler::OnLoadError.

See also Issue #74 that displays a friendly error message for other case.

Original issue reported on code.google.com by `czarek.t...@gmail.com` on 7 Oct 2014 at 4:32

**Comment by okobsamoht** [comment]
hi! i'm having the same problem!

**Comment by timint** [comment]
I got this error with both the previous version and the v130 test release in Windows 11 24H2. The previous version did work up to a point in time. I suspect it could be related to an update of a .net framework runtime or similar? Any clues?

```log
- WARNING: spawn_process: CreateProcess(php\php-cgi.exe "www\index.php"): 193
- 14:54:51.766 INFO: GET 500 /
```

```log
Error 500: Internal Server Error
Cannot spawn CGI process [www\/index.php]: No such device or address
```

**Comment by cztomczak** [comment]
@timint Try launching php-cgi.exe process manually from command line and see if it throws any errors.

**Comment by timint** [comment]
Thank you for replying.
No errors, I get the rendered HTML output.

**Comment by cztomczak** [comment]
@timint Are you sure your www/ and php/ folders are correct and settings are proper? Can you download default unmodified binaries once again and test them?

From your logs I can see error code 193 for spawn_process which means "is not a valid win32 app" (googled), so something wrong is with the php-cgi.exe binary, or maybe path is wrong?

**Comment by timint** [comment]
I think I am on to something. The v130 download works when executed from `C:\Users\Me\Downloads\phpdesktop-chrome-130.0-php-8.3`.

But it doesn't when I copy it to `D:\Projects\Folder with space char\phpdesktop-chrome-130.0-php-8.3`. So could it be the space character in the path? Funny thing is it used to work from this path until it just one day stopped.

**Comment by cztomczak** [comment]
Hmm looks like a bug. Please create an issue for that.

PHP doesn't support paths with unicode characters [1], but spaces should work fine.

[1] https://github.com/cztomczak/phpdesktop/wiki/Knowledge-Base#error-when-application-is-placed-in-a-path-that-contains-unicode-characters

**Comment by timint** [comment]
There are only a-z chars in my path. The settings file have the relative path `"cgi_interpreter": "php/php-cgi.exe",` and the log window shows this:

```log
- 15:13:38.625 WARNING: spawn_process: CreateProcess(D:\Projects\Folder with space char\phpdesktop-chrome-130.0-php-8.3\php\php-cgi.exe "D:\Projects\Folder with space char\phpdesktop-chrome-130.0-php-8.3\www\index.php"): 193
```

The path to .exe does not wrap around quotes ". But maybe that's just the way the debug text is output?

**Comment by cztomczak** [comment]
You are right, the quotes around path are missing. The source code in question is here: https://github.com/cztomczak/phpdesktop/blob/chrome130/mongoose.c#L1378

I am not sure how it worked earlier. I recall installing phpdesktop to program files which has a space and it worked fine in the past hmm.

Please create an issue in the tracker.
